### DISCUSSION TITLE
Issue with Next.js Project Including third party react-native Library

### DISCUSSION THREAD

**Discussion by iduspara-ftip** [vercel/next.js#55903] [Help]
### Summary

Hello everyone! 👋

I recently added a third-party package to my Next.js project and encountered an unexpected error related to the `react-native` library. Here's the error I encountered:

```
./node_modules/react-native/Libraries/Pressability/PressabilityDebug.js
Module parse failed: Unexpected token (11:12)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|  */
| 
> import type {ColorValue} from '../StyleSheet/StyleSheet';
| 
| import View from '../Components/View/View';

Import trace for requested module:
./node_modules/react-native/Libraries/Pressability/PressabilityDebug.js
```

For troubleshooting context, here's my configuration and setup:

### `next.config.js`
```javascript
/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: (config) => {
    config.resolve.alias = {
      ...(config.resolve.alias || {}),
      // Transform all direct `react-native` imports to `react-native-web`
      "react-native$": "react-native-web",
    };
    config.resolve.extensions = [
      ".js",
      ".jsx",
      ".ts",
      ".tsx",
      ...config.resolve.extensions,
    ];
    return config;
  },
};
module.exports = nextConfig;
```

### `package.json`
```json
{
  "name": "nextjs-tsx",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "next": "latest",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-native": "^0.72.4",
    "react-native-gesture-handler": "^2.13.1",
    "react-native-reanimated": "^3.5.4",
    "react-native-svg": "^13.14.0",
    "react-native-web": "^0.18.9"
  },
  "devDependencies": {
    "@types/node": "20.6.4",
    "@types/react": "18.2.22",
    "typescript": "5.2.2"
  }
}
```

### `tsconfig.json`
```json
{
  "compilerOptions": {
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "plugins": [
      {
        "name": "next"
      }
    ]
  },
  "include": ["next-env.d.ts", ".next/types/**/*.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

Has anyone else faced this problem before? How can I resolve it?

I would greatly appreciate any guidance or pointers on this issue. If more details are needed from my side, please let me know.

Thank you for your help!

**Comment by jkhaui** [comment]
I probably won't have time to setup a local repro and debug, but hopefully can still point you in the right direction

I highly recommend that you check out the tamagui library if you're not already familiar with it. It has some very clever workarounds to get react-native-web playing nicely with Next.js (you could download one of their starters or see how their next config is setup).

They've even forked react-native into a `react-native-web-lite` [package](https://github.com/tamagui/tamagui/tree/master/packages/react-native-web-lite) which is probably a better integration for your Next project. Your issue seems like it could be an issue with RN using flow as its type-checking system, but with RNWL Flow types are stripped in favour of TypeScript which might fix your issue.

Let me know how it goes. I've also been trying to create a smooth web integration with react-native (albeit with Astro) and it's definitely been a challenge

**Reply by iduspara-ftip** [reply]
Is there a another solution beside Tamagui, i would like to have the Setup minimal as possible.
